src/api/java/com/minecolonies/api/colony/managers/interfaces/IBuildingManager.java

```diff
@@ -77,6 +78,32 @@
      */
     IBuilding getBuilding(BlockPos pos);
 
+    /**
+     * Get the leisure site positions.
+     * @return the list.
+     */
+    List<BlockPos> getLeisureSites();
+
```

Comment:
Split apart into a leisure manager.
Leisure sites are not directly related to buildings so they should be split apart.

Reply:
I don't fully agree on this. The leisure sites function pretty exactly like buildings because.
a) Some buildings are leisure sites
b) Leisure sites use decoration controllers which are very similar to buildings.
c) Due to a) we query several building related things with this

Reply:
The key words here being **very similar but not equal **

Leisure sites are not buildings. Period.
Move it to a separate API. It would have taken you just as long to do that as it took you to write that message...

Reply:
Then I rename the manager to "RegisteredStructureManager" because it also takes care of fields forever. + Leisure is a mix of buildings and deco. It makes sense to have them together.